internal/podman/config.go

```diff
@@ -0,0 +1,170 @@
+package podman
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+
+	"github.com/docker/docker/pkg/homedir"
+	"github.com/pelletier/go-toml"
+)
+
+// env vars
+// XDG_CONFIG_HOME
+// XDG_RUNTIME_DIR
+// CONTAINER_SSHKEY
+
+// {env vars} -> {config path} -> address
+
+type containersConfig struct {
+	Engine engine `toml:"engine"`
+}
+
+type engine struct {
+	ActiveService       string                        `toml:"active_service"`
+	ServiceDestinations map[string]serviceDestination `toml:"service_destinations"`
+}
+
+type serviceDestination struct {
+	URI      string `toml:"uri"`
+	Identity string `toml:"identity"`
+}
+
+func findUnixAddressFromFile(path string) string {
+	cc, err := parseContainerConfig(path)
+	if err != nil {
+		return ""
+	}
+
+	if cc == nil {
+		return ""
+	}
+
+	return findUnixAddress(*cc)
+}
+
+func findDestinationOfType(cc containersConfig, ty string) *serviceDestination {
+	// always attempt what the config says is the current service
+	if destination, ok := cc.Engine.ServiceDestinations[cc.Engine.ActiveService]; ok {
+		if isScheme(destination.URI, ty) {
+			return &destination
+		}
+	}
+
+	// fallback to looking at all configured services if the active service is not found or is not unix
+	for _, destination := range cc.Engine.ServiceDestinations {
+		if destination.URI == "" {
+			continue
+		}
+		if isScheme(destination.URI, ty) {
+			return &destination
+		}
+	}
+	return nil
+}
+
+func findSSHConnectionInfoFromFile(path string) (string, string) {
+	cc, err := parseContainerConfig(path)
+	if err != nil {
+		return "", ""
+	}
+
+	if cc == nil {
+		return "", ""
+	}
+
+	return findSSHConnectionInfo(*cc)
+}
+func findSSHConnectionInfo(cc containersConfig) (string, string) {
+	dest := findDestinationOfType(cc, "ssh")
+	if dest == nil {
+		return "", ""
+	}
+
+	return dest.URI, dest.Identity
+}
+
+func findUnixAddress(cc containersConfig) string {
+	dest := findDestinationOfType(cc, "unix")
+	if dest == nil {
+		return ""
+	}
+	return parseUnixAddress(dest.URI)
+}
+
+func parseUnixAddress(uri string) string {
+	u, err := url.Parse(uri)
+	if err != nil {
+		return ""
+	}
+
+	if u.Scheme == "unix" {
+		return fmt.Sprintf("unix://%s", u.Path)
+	}
+	return ""
+}
+
+func isScheme(uri, scheme string) bool {
+	u, err := url.Parse(uri)
+	if err != nil {
+		return false
+	}
+
+	return u.Scheme == scheme
+}
+
+func parseContainerConfig(path string) (*containersConfig, error) {
+	configBytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var cc containersConfig
+	if err := toml.Unmarshal(configBytes, &cc); err != nil {
+		return nil, err
+	}
+	return &cc, nil
+}
+
+var (
+	// _configPath is the path to the containers/containers.conf
+	// inside a given config directory.
+	_configPath = "containers/containers.conf"
```

Comment:
- why prefix with `_`?
- This is better served as a `const`

Reply:
I agree on all the above, however Golang doesn't allow the use of filapath.Join to set constants, because go must know the const value during compilation time. So I am declaring it as a var with filepath.Join